### PR TITLE
Invoke-DbaDatabaseShrink - speed and other fixes

### PR DESCRIPTION
**Pull request by Zhunya**
## Type of Change

 - [X] Bug fix (non-breaking change, fixes #<enter issue number>)
 - [ ] New feature (non-breaking change, adds functionality)
 - [ ] Breaking change (effects multiple commands or functionality)
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] Nunit test is included
 - [ ] Documentation
 - [ ] Build system
 

### Purpose

Make Invoke-DbaDatabaseShrink faster

### Approach

Removed SMO enumeration, faster T-SQL

### CODE REVIEW

**Comment by potatoqualitee** [comment]
Thank you, @Zhunya ! Unfortunately, there are some failures on the command's test.

**Comment by Zhunya** [comment]
Its error on `Multiple ambiguous overloads found for "Round" and the argument count: "2".`
I have no idea how to fix this.

**Comment by potatoqualitee** [comment]
i believe this means that it is expecting a single value, but it's receiving multiples. One way is to SUM/GROUP up in T-SQL so that only one result is returned.

**Comment by potatoqualitee** [comment]
may need to do `($dataRow.max_fragmentation_in_percent)[0]`

File: functions/Invoke-DbaDatabaseShrink.ps1 [hunk 1]
```diff
@@ -145,24 +145,15 @@ function Invoke-DbaDatabaseShrink {
...
-        $sqlTop1 = "SELECT top 1
-                indexstats.avg_fragmentation_in_percent
-                FROM sys.dm_db_index_physical_stats (DB_ID(), NULL, NULL, NULL, NULL) AS indexstats
-                INNER JOIN sys.tables dbtables on dbtables.[object_id] = indexstats.[object_id]
-                INNER JOIN sys.schemas dbschemas on dbtables.[schema_id] = dbschemas.[schema_id]
-                INNER JOIN sys.indexes AS dbindexes ON dbindexes.[object_id] = indexstats.[object_id]
-                AND indexstats.index_id = dbindexes.index_id
-                WHERE indexstats.database_id = DB_ID()
-                ORDER BY indexstats.avg_fragmentation_in_percent desc"
+                    avg(indexstats.avg_fragmentation_in_percent) as [avg_fragmentation_in_percent],
```
Comment (wsmelton) [inline functions/Invoke-DbaDatabaseShrink.ps1]: Why are we averaging a value that is already averaged?

**Comment by wsmelton** [comment]
Which to note the reason those messages are not clear is because this command does not use the messaging system properly....it locks in the inner exception for try/catch blocks and will never output the proper message.

File: functions/Invoke-DbaDatabaseShrink.ps1 (hunk 1 above)
Comment (Zhunya) [reply functions/Invoke-DbaDatabaseShrink.ps1]: Fixed T-SQL =)
I get avg fragmentation in all database indexes

**Comment by Zhunya** [comment]
@potatoqualitee

Error in test:
```
Exception calling "LeaveTestGroup" with "2" argument(s): "TestGroups stack corrupted:  Expected name/hint of 'C:\github\dbatools\tests\Read-DbaTraceFile.Tests.ps1','Script'.  Found 'Read-DbaTraceFile Integration Tests', 
'Describe'."
At C:\Program Files\WindowsPowerShell\Modules\Pester\4.3.1\Pester.psm1:870 char:21
+ ...                    $pester.LeaveTestGroup($testScript.Path, 'Script')
+                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : ScriptMethodRuntimeException```

**Comment by potatoqualitee** [comment]
yay theres only indent problems. you can ignore all others. appveyor is shit right now.

**Comment by potatoqualitee** [comment]
i will merge and fix 👍 thank you for the perf enhancement!

**Pull request merged**